lua/pac3/core/client/parts/animation.lua

```diff
@@ -244,59 +244,55 @@ end
 function PART:UpdateAnimation(ent)
 	if not self.random_seqname then return end
 
-	local seq = ent:LookupSequence(self.random_seqname) or 0
-
-	local duration = 0
+	local seq, duration = ent:LookupSequence(self.random_seqname)
+	
 	local count = ent:GetSequenceCount() or 0
-	if seq >= 0 and seq < count and count > 0 then
-		duration = ent:SequenceDuration(seq)
-	else
+	if seq < 0 or seq >= count then
 		-- It's an invalid sequence. Don't bother
 		return
 	end
+	
+	local min = self.Min
+	local max = self.Max
```

Comment:
You could also do `local length = max - min` to optimize away the `max - min` in the later code